### DISCUSSION TITLE
Full page refresh on a dynamic route link instead of push in router state

### DISCUSSION THREAD

**Discussion by umakantp** [vercel/next.js#12667] [Help]
If there is dynamic link on the page and on click of that link, the new page is full page load instead of loading as is like router.push way.

## To Reproduce

1. `git clone https://github.com/umakantp/next-routes-bug`
2. `cd next-routes-bug && npm install && npm run dev`
3. open http://localhost:3000 on the browser with developer tools open in network tab.
4. Click the link `It reloads full page` on the page.
5. Observer the network tab of developer tools, it is full page reload.

## Expected behaviour

It should load the dynamic route as is like how SPA page open (like how route.push works)

## System information

- OS: mac OS
- Browser: chrome
- Version of Next.js: 9.3.6
- Version of Node.js: v13.8.0
- Version of React: 16.13.1

## Additional context

This bug has been happening on our live website as well. It doesn't happen when the route is not dynamic.

**Comment by umakantp** [comment]
I have seen this happening on few other websites listed here on https://github.com/zeit/next.js/discussions/10640 

I wonder why no one is talking about it?

**Comment by timneutkens** [comment]
Your usage of Link is incorrect:
https://github.com/umakantp/next-routes-bug/blob/master/pages/index.js#L9

Should be:

```jsx
<Link href="/test/[id]" as="/test/1">
```
